cmake/format.cmake

```diff
@@ -0,0 +1,67 @@
+add_custom_target(
+    rust-format-fix
+    COMMAND sed -i -e 's/ffi_panic_boundary! {/if true {/g' src/*.rs
+    COMMAND cargo fmt
+    COMMAND sed -i -e 's/if true {/ffi_panic_boundary! {/g' src/*.rs
+    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+)
+
+add_custom_target(
+    rust-format-check
+    COMMAND sed -i -e 's/ffi_panic_boundary! {/if true {/g' src/*.rs
+    COMMAND cargo fmt --check
+    COMMAND sed -i -e 's/if true {/ffi_panic_boundary! {/g' src/*.rs
+    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+)
+
+add_custom_target(
+    cmake-format-fix
+    COMMAND
+        gersemi --no-warn-about-unknown-commands -i CMakeLists.txt
+        tests/CMakeLists.txt cmake/*.cmake
+    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+)
+
+add_custom_target(
+    cmake-format-check
+    COMMAND
+        gersemi --no-warn-about-unknown-commands -c CMakeLists.txt
```

Comment:
Seems reasonable. I went ahead and did that in a separate commit in case anyone thinks it's more clunky than valuable to duplicate the function defns.